src/store/accounts/index.ts

```diff
@@ -174,13 +174,29 @@ export const allAccountDataRefreshState = atom(null, (get, set) => {
 // the balances of the current account's address
 export const accountBalancesState = atom<AllAccountData['balances'] | undefined>(get => {
   const balances = get(currentAccountDataState)?.balances;
-  const stxBalance = get(accountInfoState)?.balance.toString(10);
+  const stxBalance = balances ? balances.stx.balance : '';
   return balances
     ? {
         ...balances,
         stx: {
           ...balances.stx,
-          balance: stxBalance || balances.stx.balance,
+          balance: stxBalance,
+        },
+      }
+    : undefined;
+});
+
+export const accountUnanchoredBalancesState = atom<
+  AllAccountData['unanchoredBalances'] | undefined
+>(get => {
+  const balances = get(currentAccountDataState)?.unanchoredBalances;
+  const stxBalance = balances ? balances.stx.balance : '';
+  return balances
+    ? {
+        ...balances,
+        stx: {
+          ...balances.stx,
+          balance: stxBalance,
```

Comment:
It'd be great if we could initialise currency values as `BigNumber` at the store level. Otherwise this is repeated many places in the codebase.

Further, we should be careful about passing around currency values as strings. A currency value should really contain two things: 
1) The currency's symbol
2) The amount of given currency

Working on a cryptocurrency exchange in the past, we had a `Money` type, that worked quite well. A very simple, immutable class with these two values. Here it'd be consumed like: `{ balance: Money.create('STX', stxBalance) }`.

Reply:
Agree indeed this will keep more information as we pass the value around. I have used math.js in the past to do something similar.
However I think we can do that in another PR since this is only related to microblocks and the necessary changes could introduce potentials regression.